tests/JsonEncodeExceptionTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Crell\ApiProblem;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test for the JSON error handling.
+ */
+class JsonEncodeExceptionTest extends TestCase
+{
+    public function testConstructorDefaults(): void
+    {
+        $exception = new JsonEncodeException();
+
+        $this->assertSame('', $exception->getMessage());
+        $this->assertSame(0, $exception->getCode());
+        $this->assertNull($exception->getPrevious());
+        $this->assertNull($exception->getJson());
+    }
+
+    public function testConstructor(): void
+    {
+        $previous = new \RuntimeException();
+        $exception = new JsonEncodeException(
+            'title',
+            2,
+            $previous,
+            'json'
+        );
+
+        $this->assertSame('title', $exception->getMessage());
+        $this->assertSame(2, $exception->getCode());
+        $this->assertSame($previous, $exception->getPrevious());
+        $this->assertSame('json', $exception->getJson());
+    }
+
+    public function testWithJsonObject(): void
+    {
+        $json = new \stdClass();
+
+        $previous = new \RuntimeException();
+        $exception = new JsonEncodeException(
+            'title',
+            2,
+            $previous,
+            $json
+        );
+
+        $this->assertSame('title', $exception->getMessage());
+        $this->assertSame(2, $exception->getCode());
+        $this->assertSame($previous, $exception->getPrevious());
+        $this->assertSame($json, $exception->getJson());
+    }
```

Comment:
I'm not sure I get the purpose of these two tests being separate.  The only difference is whether the JSON value provided (which is not actually JSON, but the supposed-JSON) is actually valid JSON, but that's not relevant to the code in question.  So why test it separately?  Seems like one of these can be removed.

Reply:
It's a unit testing, so I tested it like that, as a unit.